firefox-ios/Storage/Rust/RustRemoteTabs.swift

```diff
@@ -68,20 +65,18 @@ public class RustRemoteTabs {
 
     public func setLocalTabs(localTabs: [RemoteTab]) -> Deferred<Maybe<Int>> {
         let deferred = Deferred<Maybe<Int>>()
-
-        queue.async {
-            let tabs = localTabs.map { $0.toRemoteTabRecord() }
-
-            if let storage = self.storage {
-                storage.setLocalTabs(remoteTabs: tabs)
-                deferred.fill(Maybe(success: tabs.count))
-            } else {
-                let error = TabsApiError.UnexpectedTabsError(reason: "Unknown error when setting local Rust Tabs")
-                deferred.fill(Maybe(failure: error as MaybeErrorType))
-            }
-        }
-
-        return deferred
+           queue.async {
+               guard let store = self.store else {
+                   deferred.fill(
+                    Maybe(failure: TabsApiError.UnexpectedTabsError(reason: "TabsStore is not available") as MaybeErrorType)
+                    )
```

Comment:
nit: prefer to have the error in a separate line similar to before for easier readability